thunder/benchmarks/targets.py

```diff
@@ -934,3 +940,37 @@ def test_lora_linear(benchmark, executor, compute_type, implementation):
     fn = executor(b.fn())
 
     benchmark_for_compute_type(compute_type, benchmark, fn, args, kwargs)
+
+
+#
+# optimizer benchmarks
+#
+
+
+@pytest.mark.parametrize(
+    "executor,",
+    [
+        thunderfx_executor,
+        torch_compile_executor,
+        torch_executor,
+    ],
+    ids=["thunderfx", "inductor", "eager"],
+)
+@parametrize_compute_type_without_backward
+@pytest.mark.parametrize(
+    "params",
+    [(64, 64), (128, 64)],
+    ids=["64x64", "128x64"],
+)
+def test_optim_functional_adam(benchmark, executor: None | Callable, params: Sequence[int], compute_type: ComputeType):
+    bench: Benchmark = AdamBenchmark(
+        params=params,
+        device="cuda:0",
+        dtype=thunder.float32,
+        requires_grad=is_requires_grad(compute_type),
+    )
+
+    jfn = executor(bench.fn())
+    args, kwargs = bench.make_batch()
+
+    benchmark_for_compute_type(compute_type, benchmark, jfn, args, kwargs)
```

Comment:
nit. when the executor is eager then this function is not jitted, therefore let's keep this `fn`
```suggestion
    fn = executor(bench.fn())
    args, kwargs = bench.make_batch()

    benchmark_for_compute_type(compute_type, benchmark, fn, args, kwargs)
```